lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateTaskAccTest.java

```diff
@@ -0,0 +1,561 @@
+    taskService.updateTask(task);
+    task2.setCustomField(CUSTOM_2, "Walter");
+    // TODO flaky test ... if speed is too high,
+    assertThatThrownBy(() -> taskService.updateTask(task2))
+        .isInstanceOf(ConcurrencyException.class)
+        .hasMessage(
+            "The entity with id '%s' cannot be updated since it has been modified while editing.",
+            task.getId());
+  }
+
+  @WithAccessId(user = "admin")
+  @WithAccessId(user = "taskadmin")
+  @TestTemplate
+  void should_UpdateTask_When_NoExplicitPermissionsButUserIsInAdministrativeRole()
+      throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    ClassificationSummary classificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService, "admin");
+    task.setClassificationKey(classificationSummary.getKey());
+
+    ThrowingCallable updateTaskCall = () -> taskService.updateTask(task);
+
+    assertThatCode(updateTaskCall).doesNotThrowAnyException();
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UpdateTaskProperties_When_ClassificationOfTaskIsChanged() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    ClassificationSummary classificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService, "admin");
+    task.setClassificationKey(classificationSummary.getKey());
+    Task updatedTask = taskService.updateTask(task);
+    updatedTask = taskService.getTask(updatedTask.getId());
+
+    assertThat(updatedTask).isNotNull();
+    assertThat(updatedTask.getClassificationSummary().getKey())
+        .isEqualTo(classificationSummary.getKey());
+    assertThat(updatedTask.getClassificationSummary()).isNotEqualTo(defaultClassificationSummary);
+    assertThat(updatedTask.getCreated()).isNotEqualTo(updatedTask.getModified());
+    assertThat(task.getPlanned()).isEqualTo(updatedTask.getPlanned());
+    assertThat(task.getName()).isEqualTo(updatedTask.getName());
+    assertThat(task.getDescription()).isEqualTo(updatedTask.getDescription());
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UpdateReadFlagOfTask_When_SetTaskReadIsCalled() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    taskService.setTaskRead(task.getId(), true);
```

Comment:
Does setTaskRead return a Task? If so, you don't need to use "getTask" in the next line